libs/community/langchain_community/chat_models/huggingface.py

```diff
@@ -42,7 +37,7 @@ class ChatHuggingFace(BaseChatModel):
     Adapted from: https://python.langchain.com/docs/integrations/chat/llama2_chat
     """
 
-    llm: Union[HuggingFaceTextGenInference, HuggingFaceEndpoint, HuggingFaceHub]
```

Comment:
a simpler solution  that i think works would be to just type this as `Any`, and then pydantic won't do any validation.

Reply:
@baskaryan this would then let the user free of using any model, even non HuggingFace ones, and get broken results: do we still want to do this?

Reply:
my sense is too permissive is better than too restrictive, especially if the restriction is a breaking one? if we really care we can add a `@root_validator` to do custom validation of the `llm` to make sure it's a valid type

Reply:
like this https://github.com/langchain-ai/langchain/pull/18693

Reply:
Your fix with the `@root_validator` works for all 3 classes! Thank a lot!